src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.cs

```diff
@@ -85,7 +87,8 @@ internal async Task InitializeStandardComponentServicesAsync(
 
         if (navigationManager != null)
         {
-            navigationManager.OnNotFound += async (sender, args) => await SetNotFoundResponseAsync(navigationManager.BaseUri);
+            navigationManager.OnNotFound += async (sender, args) =>
+                await GetErrorHandledTask(SetNotFoundResponseAsync(navigationManager.BaseUri, args));
```

Comment:
You don't need the async/await pair here. Just `(sender, args) => { _ = GetErrorHandledTask(SetNotFoundResponseAsync(navigationManager.BaseUri, args); }`